core/src/main/java/org/infinispan/marshall/exts/CacheEventTypeExternalizer.java

```diff
@@ -0,0 +1,35 @@
+package org.infinispan.marshall.exts;
+
+import org.infinispan.commons.marshall.AbstractExternalizer;
+import org.infinispan.commons.util.Util;
+import org.infinispan.marshall.core.Ids;
+import org.infinispan.notifications.cachelistener.event.Event;
+
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+import java.util.Set;
+
+public final class CacheEventTypeExternalizer extends AbstractExternalizer<Event.Type> {
```

Comment:
`ClusterEvent.Externalizer.readObject()` should use `MarshallUtil.marshallEnum/unmarshallEnum` instead.